src/components/App.js

```diff
@@ -0,0 +1,67 @@
+/* eslint-disable react/jsx-no-bind */
+import React, { useState, useEffect } from "react";
+import "../styles/App.css";
+import getForecast from "../requests/getForecast";
+import LocationDetails from "./LocationDetails";
+import ForecastSummaries from "./ForecastSummaries";
+import ForecastDetails from "./ForecastDetails";
+import SearchForm from "./SearchForm";
+
+function App() {
+  const [forecasts, setForecasts] = useState([]);
+  const [location, setLocation] = useState({
+    city: "Hi",
+    country: "want to search the weather?",
+  });
+  const [selectedDate, setSelectedDate] = useState(0);
+  const [searchText, setSearchText] = useState("");
+  const [errorMessage, setErrorMessage] = useState("");
+
+  const selectedForecast = forecasts.find(
+    (forecast) => forecast.date === selectedDate
+  );
+
+  useEffect(() => {
+    getForecast(setSelectedDate, setForecasts, setLocation);
+  }, []);
+
+  const handleCitySearch = () => {
+    getForecast(
+      searchText,
+      setSelectedDate,
+      setForecasts,
+      setLocation,
+      setErrorMessage
+    );
+  };
+
+  const handleForecastSelect = (date) => {
+    setSelectedDate(date);
+  };
+
```

Comment:
You could destructure the city and the country here like this:
`const {city, country} = location`